contrib/nix/local-build-shell.nix

```diff
@@ -0,0 +1,28 @@
+# Refer to README.md for how to invoke this.
+
+{ pkgs ? import <nixpkgs> {} }:
+pkgs.mkShell {
+  nativeBuildInputs = with pkgs.buildPackages; [
+    autoconf
+    cmake
+    git
+    libsodium
+    libunwind
+    openssl
+    protobuf
+    python3
+    readline
+    unbound
+    zeromq
+  ]  ++ lib.optionals stdenv.isDarwin [
+    darwin.apple_sdk.frameworks.IOKit
+  ];
+
+  buildInputs = with pkgs.buildPackages; [
+    boost
+  ];
+
+  cmakeFlags = [
+    "-DReadline_ROOT_DIR=${pkgs.buildPackages.readline.dev}"
```

Comment:
As a side note. This PR [1] will fix this requirement. 

1. https://github.com/monero-project/monero/pull/9445